tests/components/CopyPaste/CopyPaste.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { IQueryResult, $$, l } from 'coveo-search-ui';
+import { createSandbox, SinonSandbox, SinonStub } from 'sinon';
+import { Mock, Fake } from 'coveo-search-ui-tests';
+import { CopyPaste } from '../../../src/components/CopyPaste/CopyPaste';
+
+describe('ResultAction', () => {
+    let sandbox: SinonSandbox;
+
+    let componentSetup: Mock.IBasicComponentSetup<CopyPaste>;
+    let result: IQueryResult;
+    let element: HTMLElement;
+    let testComponent: CopyPaste;
+    let testOptions: Mock.AdvancedComponentSetupOptions;
+    let writeTextStub: SinonStub<[string], Promise<void>>;
+
+    beforeAll(() => {
+        sandbox = createSandbox();
+    });
+
+    beforeEach(async () => {
+        result = Fake.createFakeResult();
+        element = document.createElement('div');
+        document.body.append(element);
+        testOptions = new Mock.AdvancedComponentSetupOptions(element);
+
+        result.title = 'some title';
+        result.clickUri = 'some uri';
+
+        componentSetup = Mock.advancedResultComponentSetup(CopyPaste, result, testOptions);
+        testComponent = componentSetup.cmp;
+        writeTextStub = sandbox.stub(navigator.clipboard, 'writeText').resolves();
```

Comment:
This does not test the IE11 code path. It would be nice to have a test case that covers IE11.

Reply:
Sadly you can't. (Trust me I try)

Reply:
Did you try spying on `document.execCommand`? Or we could add an event like `onCopy` that would contain the copied information. This way we would be able to easily validate that the right data is sent without having to deal directly with the browser.